rust/http-test-server/Cargo.toml

```diff
@@ -0,0 +1,9 @@
+[package]
+name = "http-test-server"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+httpmock = "0.7.0"
```

Comment:
Just saw that httpmock has [an open issue](https://github.com/alexliesenfeld/httpmock/issues/82) about depending (indirectly) on OpenSSL.
We don't need any SSL support in the http test server but I still need to check if we can build successfully with alternative SSL libraries (GnuTLS, ?).

Reply:
I'm having some trouble setting up a system containing just GnuTLS (specifically, no OpenSSL).
For example, on Ubuntu and Alpine Linux, just installing a curl-gnutls-dev package will already pull in libssl.
Removing the openssl / libssl packages caused the removal of a lot of other packages, even removing a few compiler packages.

(note, package names might be slightly incorrect as I lost the command history of my experimentation)

Reply:
According to that issue, httpmock 0.7.0 (which we use) doesn't depend on OpenSSL by default, because the `remote` feature is disabled. The updated Cargo.lock doesn't contain any mention of SSL too. So we should be good?

Reply:
> So we should be good?

Indeed, I didn't notice before. thanks!
That makes this PR ready for review/merging.